### PR TITLE
FYLE-2891u5w-preview

### PR DESCRIPTION
**Pull request by Kulsekar4**

### CODE REVIEW

File: src/admin/paths/admin@statements@preview.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+post:
+  tags:
+    - Statements
+  summary: Preview of the uploaded statement
+  description: |
+    This API helps in converting the statement with the mapping chosed and helps the user to preview it, this doesn't save the statement
+  operationId: statements_prevew
```
Comment (modernwarfareuplink) [inline src/admin/paths/admin@statements@preview.yaml:7]: preview

File: src/components/schemas/statements.yaml [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+          description: |
+            Specifies the currency used in the statement.
+          example: USD
+        name:
+          type: string
+          description: |
+            Specifies the mapping name.
+          example: USD
+          nullable: true
+    ignored_line_items:
```
Comment (modernwarfareuplink) [inline src/components/schemas/statements.yaml:77]: How will the client find the id of the statement line items during the preview stage ? @Kulsekar4

File: src/components/schemas/statements.yaml [hunk 3]
```diff
@@ -0,0 +1,151 @@
+preview_in:
+  type: object
+  properties:
+    id:
+      allOf:
+        - $ref: './fields.yaml#/id_string'
+      example: stmtadfdsdf
+    statement_mapping:
```
Comment (modernwarfareuplink) [inline src/components/schemas/statements.yaml:8]: add `required` for this object too @Kulsekar4

File: src/components/schemas/statements.yaml [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+          example: amount
+        card_number_column:
+          type: string
+          description: |
+            Specifies the mapping of card number on the statement.
+          example: account number
+        vendor_column:
+          type: string
+          description: |
+            Specifies the vendor of the statement
```
Comment (modernwarfareuplink) [inline src/components/schemas/statements.yaml:41]: Specifies the mapping of vendor on the statement.

File: src/components/schemas/statements.yaml [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        bank_name:
+          type: string
+          description: |
+            Specifies the bank name of the statement
+          example: American Express
+        is_debit_positive:
+          type: boolean
+          description: |
+            Specifies the representation of the the amount in the statement, if true then the amount represented by |
+            negative sign are credits, and if false then the amount represented by negative sign are debits.
```
Comment (modernwarfareuplink) [inline src/components/schemas/statements.yaml:64]: Specifies the representation of the the amount in the statement, if true then the amount represented by positive sign are debits, and if false then the amount represented by positive sign are credits.

File: src/components/schemas/statements.yaml [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+          description: |
+            Specifies the mapping name.
+          example: USD
+          nullable: true
+    ignored_line_items:
+      type: array
+      items: 
+        type: string
+      description: >
+        This specifies list of line items to be ignored in the statement.
```
Comment (modernwarfareuplink) [inline src/components/schemas/statements.yaml:82]: statement line items

File: src/components/schemas/statements.yaml [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    - id
+    - statement_mapping
+    - ignored_line_items
+
+preview_out:
+  properties:
+    transactions:
+      type: array
+      items: 
+        type: object
```
Comment (modernwarfareuplink) [inline src/components/schemas/statements.yaml:95]: add `required` for this object too

File: src/components/schemas/statements.yaml [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+        properties:
+          description:
+            type: string
+            description: |
+              Specifies the header name of the description column in the statement.
```
Comment (modernwarfareuplink) [inline src/components/schemas/statements.yaml:100]: This is not the header name @Kulsekar4 

Can use something like below with some changes.
`Description of the transaction in the statement as per description column mapping.`

File: src/components/schemas/statements.yaml [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+              Specifies the foreign amount on the statement.
+            nullable: true
+            example: 48
+          foreign_currency:
+            type: string
+            description: |
+              Specifies the foreign currency on the statement.
+            nullable: true
+            example: USD
+          transaction_id:
```
Comment (modernwarfareuplink) [inline src/components/schemas/statements.yaml:144]: `transaction_type` is missing @Kulsekar4

File: src/components/schemas/statements.yaml (hunk 2 above)
Comment (Kulsekar4) [reply src/components/schemas/statements.yaml:84]: In the very first api `statements/upload` the `statement_line_item_id` which is duplicate would be sent sir.
And here the user can pass those id's

File: src/components/schemas/statements.yaml (hunk 7 above)
Comment (Kulsekar4) [reply src/components/schemas/statements.yaml:102]: Added sir

File: src/components/schemas/statements.yaml (hunk 9 above)
Comment (Kulsekar4) [reply src/components/schemas/statements.yaml:151]: Added sir

**Pull request closed without merging**